docs/recipes/precompiling-with-webpack.md

```diff
@@ -29,3 +37,226 @@ module.exports = {
+const glob = require('glob');
+
+const nodeExternals = require('webpack-node-externals');
+
+module.exports = {
+	target: 'node',
+	entry: glob.sync('./test/**/*.js'),
+	output: {
+		path: path.resolve(__dirname, '_build'),
+		filename: 'tests.js'
+	},
+	externals: [nodeExternals()],
+	module: {
+		rules: [{
+			test: /\.(js|jsx)$/,
+			use: {
+				loader: 'babel-loader',
+				options: { cacheDirectory: true }
+			}
+		}]
+	}
+};
+```
+
+<details>
+<summary>Error report comparison</summary>
+
+```
+# Before
+  aggregations-test » cardinality-agg » sets precision_threshold option
+  E:\Projects\repos\elastic-builder\test\_macros.js:167
+
+   166:         const expected = getExpected(keyName, recursiveToJSON(propValue));
+   167:         t.deepEqual(value, expected);
+   168:     }
+
+  Difference:
+
+      Object {
+        my_agg: Object {
+          cardinality: Object {
+    -       precision_threshol: 5000,
+    +       precision_threshold: 5000,
+          },
+        },
+      }
+
+# After
+  sets precision_threshold option
+  E:\Projects\repos\elastic-builder\_build\tests.js:106
+
+   105:                     column: 21
+   106:                 }
+   107:             },
+
+  Difference:
+
+      Object {
+        my_agg: Object {
+          cardinality: Object {
+    -       precision_threshol: 5000,
+    +       precision_threshold: 5000,
+          },
+        },
+      }
+
+```
+
+</details>
+
+#### Multiple entry files
+We can ask webpack to generate multiple entry files. This helps retain file names so that error reports are easy to interpret. But each entry file gets it's own copy of the source files. This results in considerably larger file sizes. This can [perform quite poorly](https://github.com/avajs/ava/pull/1385#issuecomment-304684047) on the first execution.
+
+###### webpack.config.js
+
+```js
+const path = require('path');
+const glob = require('glob');
+
+const nodeExternals = require('webpack-node-externals');
+
+const testFiles = glob.sync('./test/**/*.js');
+
+const entryObj = {};
+const len = testFiles.length;
+for (let idx = 0; idx < len; idx++) {
+	const file = testFiles[idx];
+	entryObj[path.basename(file, path.extname(file))] = file;
+}
+
+module.exports = {
+	target: 'node',
+	entry: entryObj,
+	output: {
+		path: path.resolve(__dirname, '_build'),
+		filename: '[name].js'
+	},
+	externals: [nodeExternals()],
+	module: {
+		rules: [{
+			test: /\.(js|jsx)$/,
+			use: {
+				loader: 'babel-loader',
+				options: { cacheDirectory: true }
+			}
+		}]
+	}
+};
+```
+
+#### Multiple entry files with externalized source files
+This is the most complicated to setup but performs quite well and also retains file names. In this approach, we use the babel cli to compile the src files but preserve file structure. Then we compile the tests with a require path rewrite. The following example is for a specific file structure. Depending on how your source and test files are organised, you might need to make changes.
```

Comment:
I'd replace "babel cli" with `babel-cli`. Rather than "compile the src file" I'd use "compile the source files". And I'd invert the sentence structure from:

> Then we compile the tests with a require path rewrite"

To:

> Require paths in tests are rewritten when compiling them in webpack.